src/components/reset-password/tests/reset-password-2fa-before-integration.test.ts

```diff
@@ -0,0 +1,224 @@
+        password: "dad",
+        "confirm-password": "",
+      })
+      .expect(function (res) {
+        const $ = cheerio.load(res.text);
+        expect($("#password-error").text()).to.contains(
+          "Your password must be at least 8 characters long and must include letters and numbers"
+        );
+      })
+      .expect(400, done);
+  });
+
+  it("should return validation error when password is amongst most common passwords", (done) => {
+    nock(baseApi).post("/reset-password").once().reply(400, { code: 1040 });
+
+    request(app)
+      .post(ENDPOINT)
+      .type("form")
+      .set("Cookie", cookies)
+      .send({
+        _csrf: token,
+        password: "password123",
+        "confirm-password": "password123",
+      })
+      .expect(function (res) {
+        const $ = cheerio.load(res.text);
+        expect($("#password-error").text()).to.contains(
+          "Enter a stronger password. Do not use very common passwords, such as ‘password’ or a sequence of numbers."
+        );
+      })
+      .expect(400, done);
+  });
+
+  it("should return error when new password is the same as existing password", (done) => {
+    nock(baseApi).post("/reset-password").once().reply(400, { code: 1024 });
+
+    request(app)
+      .post(ENDPOINT)
+      .type("form")
+      .set("Cookie", cookies)
+      .send({
+        _csrf: token,
+        password: "p@ssw0rd-123",
+        "confirm-password": "p@ssw0rd-123",
+      })
+      .expect(function (res) {
+        const $ = cheerio.load(res.text);
+        expect($("#password-error").text()).to.contains(
+          "You are already using that password. Enter a different password"
+        );
+      })
+      .expect(400, done);
+  });
+
+  it("should return validation error when no numbers present in password", (done) => {
+    request(app)
+      .post(ENDPOINT)
+      .type("form")
+      .set("Cookie", cookies)
+      .send({
+        _csrf: token,
+        password: "testpassword",
+        "confirm-password": "testpassword",
+      })
+      .expect(function (res) {
+        const $ = cheerio.load(res.text);
+        expect($("#password-error").text()).to.contains(
+          "Your password must be at least 8 characters long and must include letters and numbers"
+        );
+      })
+      .expect(400, done);
+  });
+
+  it("should return validation error when password all numeric", (done) => {
+    request(app)
+      .post(ENDPOINT)
+      .type("form")
+      .set("Cookie", cookies)
+      .send({
+        _csrf: token,
+        password: "222222222222222",
+        "confirm-password": "222222222222222",
+      })
+      .expect(function (res) {
+        const $ = cheerio.load(res.text);
+        expect($("#password-error").text()).to.contains(
+          "Your password must be at least 8 characters long and must include letters and numbers"
+        );
+      })
+      .expect(400, done);
+  });
+
+  it("should redirect to MFA step when valid password entered", (done) => {
```

Comment:
Is this what this test actually does? It looks to me like this is testing a redirect to /auth-code, which is the end of the journey.